apps/stepcounter/stepcounter.star

```diff
@@ -0,0 +1,420 @@
+    "Sunday": 0,
+    "Monday": 1,
+    "Tuesday": 2,
+    "Wednesday": 3,
+    "Thursday": 4,
+    "Friday": 5,
+    "Saturday": 6,
+}
+
+# Other Conversions for obtaining Historical Day
+SECONDS_IN_A_DAY = 3600 * 24
+SECONDS_IN_A_WEEK = SECONDS_IN_A_DAY * 7
+
+# This is the Google API URL to pull data from Google FIT.   This particular API is used to aggregate Fitness data over a given time range
+# This program uses step count data (defined in the request body below)
+
+GOOGLEFIT_DATASET_URL = "https://www.googleapis.com/fitness/v1/users/me/dataset:aggregate"
+
+# Note: Refresh Token during development was pulled using the Google Playground Console.  Requires User Approval sequence via browser.
+# these tokens are reputed to be "permanent" for now, so pasting above.  Else, Need to set up a simple Web server to execute the Google Auth code/token exchange sequence.
+#
+# A token can be obtained manually via the Playground console (need to ensure the console is configured with TidByt project in your Google developers account - dont forget to give
+# Permission for you (user) to access the app while it's in test state.
+# Alternately, you may retrieve an Authorization Code via a browser link with redirect to localhost (doesn't really do anything more than the console but was a fun diversion)
+# URL looks like: https://accounts.google.com/o/oauth2/v2/auth?client_id=XXXX.apps.googleusercontent.com&redirect_uri=127.0.0.1&scope=https://www.googleapis.com/auth/userinfo.profile&response_type=code
+# Need to allow the redirect_uri in the credentials console.
+
+# Google API Call and Body to Get a New Access Token.   These expire every 60 minutes.   However, there doesn't seem to be any harm in fetching more often.
+# This script is intended to be invoked once every 30 minutes to refresh step count.
+
+GOOGLE_OAUTH_TOKEN_REFRESH_URL = "https://oauth2.googleapis.com/token"
+GOOGLE_OAUTH_USER_AUTH_URL = "https://accounts.google.com/o/oauth2/v2/auth"
+
+# Random Hash Strings to store Secrets required by Google Oauth (for input to Tidbyt "Secrets" functions)
+# Note - app name is "stepcounter"
+CLIENT_ID_HASH = "AV6+xWcE+SIkQgnPgHzViV78GTRoxpMjlccjdOSUxNRSaBunq5fHKq5xp3sMlKVtYs1V9ZFwBUWg79Pgw+Y3mXoPB5q9AuBVN9bjgND9YpZ9dn3crPs7saefSsj+Mx4K8QUjQgzwLm68+qfWCCtQO419dnPJANjmjXuCrEk02RGw1q3DTRlmaF+Fh+Nf8PRl7wD7Vpfv++8I+WjUlqlhRviULKbMJkyRlZMuBrai"
+CLIENT_SECRET_HASH = "AV6+xWcEb1T8b5kw+ugpxOQ55oRdM9Ox+/PxPSm7V3VTC7NtCrMJXsMA/oozP2Eu8yKUnuDO2jmRB87tsr9ffX1sIkUTdLbftv4swDYku77yz79AJb31q0IRS/gxVkeYuLdgwIt2wqFX6Xrqve2t3wvouaI2WIrpH7U9YzWwc1Iwuv8+6NcKjH0="
+
+# Default (bogus) client ID and Secrets to keep the run time Env happy when running in Deubg Mode
+CLIENT_ID_DEFAULT = "123456"
+CLIENT_SECRET_DEFAULT = "78910"
+
+# Production Code - runs in the Tidbyt production environment.  In Deubg mode with Serve need to hard code valid ID and SECRET here instead (since HASHes above are only valid in production env
+GOOGLE_CLIENT_ID = secret.decrypt(CLIENT_ID_HASH)
```

Comment:
FYI - we'll replace these with Tidbyt provided information post merge.

Reply:
Yup, makes sense.